From 17ad3b52bad5dc0b02e748a70689e3d383d61ad1 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:46:26 -0400
Subject: [pci] Use request_resource_conflict() instead of insert_ for bus numbers

Message-id: <20140806034626.1438.73827.stgit@gir.stowe>
Patchwork-id: 87431
O-Subject: [RHEL7.1 PATCH 22/69] PCI: Use request_resource_conflict() instead of insert_ for bus numbers
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: ced04d15519a15d38b46162b94a1f26b4022116e

commit ced04d15519a15d38b46162b94a1f26b4022116e
Author: Andreas Noever <andreas.noever@gmail.com>
Date:   Thu Jan 23 21:59:24 2014 +0100

    PCI: Use request_resource_conflict() instead of insert_ for bus numbers

    If a conflict happens during insert_resource_conflict() and all conflicts
    fit within the newly inserted resource then they will become children of
    the new resource. This is almost certainly not what we want for bus
    numbers.

    Signed-off-by: Andreas Noever <andreas.noever@gmail.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 590afb1..fc19d73 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1903,7 +1903,7 @@ int pci_bus_insert_busn_res(struct pci_bus *b, int bus, int bus_max)
   res->flags |= IORESOURCE_PCI_FIXED;
  }
 
- conflict = insert_resource_conflict(parent_res, res);
+ conflict = request_resource_conflict(parent_res, res);
 
  if (conflict)
   dev_printk(KERN_DEBUG, &b->dev,
-- 
1.7.1